File: csharp/resistor-color-duo/ResistorColorDuo.cs
public static class ResistorColorDuo
{
    private static Dictionary<string, int> colorsToValuesDict = new()
    {
        {"black", 0},
        {"brown", 1},
        {"red", 2},
        {"orange", 3},
        {"yellow", 4},
        {"green", 5},
        {"blue", 6},
        {"violet", 7},
        {"grey", 8},
        {"white", 9},
    };


    public static int Value(string[] colors)
        => colorsToValuesDict[colors[0]] * 10 + colorsToValuesDict[colors[1]];
}

File: java/resistor-color-duo/src/main/java/ResistorColorDuo.java
import java.util.List;

class ResistorColorDuo {
    List<String> colorsList = List.of(
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white"
    );

    int value(String[] colors) {
        return this.colorsList.indexOf(colors[0]) * 10 +
            this.colorsList.indexOf(colors[1]);
    }
}
